lib/services/elasticsearch.js

```diff
@@ -73,6 +75,15 @@ class ElasticSearch extends Service {
       return Bluebird.resolve(this);
     }
 
+    this.config.dynamicMapping = this.config.dynamicMapping.toString();
+    if (this.dynamicMappingPolicies.indexOf(this.config.dynamicMapping) === -1) {
+      throw new KuzzleInternalError(`Wrong configuration value '${this.config.dynamicMapping}' for server.db.dynamicMapping. Allowed values are the following string '${this.dynamicMappingPolicies.join('\', \'')}'.`);
+    }
+
+    if (process.env.NODE_ENV === 'production' && this.config.dynamicMapping !== 'strict') {
+      this.kuzzle.pluginsManager.trigger('log:warn', 'You are not using the \'strict\' policy for collection mappings.\nYou should strictly define your mapping before sending data to prevent Elasticsearch default mapping types.');
```

Comment:
Unless I misunderstood what the `false` value entails,  I believe that only the `true` value warrants a warning in production: silently ignoring new fields seems a perfectly viable option to me.

Reply:
I'm not quite sure because it can lead to confusion if you create document with field that are gonna be ignored by Elasticsearch in subsequent search queries.

Reply:
Since that's not the default value, I don't see the problem.